packages/experimental/Icon/src/Icon.tsx

```diff
@@ -55,16 +55,20 @@ function renderSvg(iconProps: IconProps) {
   const viewBox = iconProps.svgSource.viewBox;
   const style = mergeStyles(iconProps.style, rasterImageStyleCache({ width: width, height: height }, [width, height])[0]);
 
+  // The svg color can be set using either style.color or iconProps.color, where style.color is preferred in case both are set.
+  const getColor = (color1: Color, color2: Color) => (color1 === undefined ? color2 : color1);
+  const color = getColor((style as any).color, iconProps.color);
```

Comment:
I feel like there's an immutable style merge function that we have that does the same ish thing. I'll defer to @falseLobster

Reply:
This change is made to supply the color to `<SvgUri />` which is used as fill value in the SVG. I didn't know how we could target a theme-token to a prop. For all the usages in the repo, the theme-tokens are merged into styles. 
So made this to have an override for the checkmark color using style. (and any future requirements of overriding SVG icon colors using style)

Reply:
Here's info on `mergeStyles`, which might be what you're thinking of: https://github.com/microsoft/fluentui-react-native/tree/3c0f5496f0daeb748c6b2daac8d0465a1d58a672/packages/framework/merge-props. There's also a README on `immutableMerge`: https://github.com/microsoft/fluentui-react-native/tree/3c0f5496f0daeb748c6b2daac8d0465a1d58a672/packages/framework/immutable-merge. Both work on objects, while the color prop is a string/`PlatformColor`, so it's a bit overkill here.

As a suggestion, the code could be simplified to `const color = style.color ? style.color : iconProps.color` without defining `getColor`.